mllib/src/main/scala/org/apache/spark/ml/feature/RFormula.scala

```diff
@@ -62,19 +77,72 @@ class RFormula(override val uid: String)
   /** @group getParam */
   def getFormula: String = $(formula)
 
-  /** @group getParam */
-  def setFeaturesCol(value: String): this.type = set(featuresCol, value)
+  override def fit(dataset: DataFrame): RFormulaModel = {
+    require(parsedFormula.isDefined, "Must call setFormula() first.")
+    // StringType terms and terms representing interactions need to be encoded before assembly.
+    // TODO(ekl) add support for feature interactions
+    var encoderStages = Seq[PipelineStage]()
```

Comment:
minor: `Seq` could be replaced by `ArrayBuffer` to avoid creating temp sequences. Then `:+=` below becomes `+=`, slightly simpler to read.